### PR TITLE
Add docking node

### PR DESCRIPTION
**Pull request by jim-v**
The docking node provides a dock service that can be invoked with a command like this:

```
rosservice call /dock "fiducial_id: 306
waypoints: '-1 Y 90,-.3 0 0 '" 
```

The robot will rotate until it either sees the target fiducial or completes one revolution.  If a fiducial is seen, it will attempt to navigate to the waypoints in sequence.  The format of a waypoint is three elements representing `x`, `y`, and `theta`. `x` and `y` are co-ordinates relative to the fiducial in meters and `theta is a heading angle in degrees.  The strings `X` and `Y` are replaced by the current `x` and `y` co-ordinates, respectively.

### CODE REVIEW

File: docking/package.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<package>
+  <name>docking</name>
+  <version>0.0.0</version>
+  <description>The docking package</description>
+
+  <maintainer email="jimv@mrjim.com">Jim Vaughan</maintainer>
+  <maintainer email="send2arohan@gmail.com">Rohan Agrawl</maintainer>
```
Comment (rohbotics) [inline docking/package.xml]: spelling of my name haha

Comment (rohbotics) [inline docking/package.xml]: Spelling of my name haha

File: docking/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+`/dock` (`std_srvs/Trigger`): invokes the docking service
+
+
+### Publications
+
+`/move_base/goal` (`move_base_msgs/MoveBaseActionGoal`): sends a movement goal
+
+### Parameters
+
+`~waypoints` a string containg a list of poses in the form `x0 y0 theta, x1 y1 theta1`, where
```
Comment (rohbotics) [inline docking/README.md]: Isn't this not a parameter?

File: docking/nodes/dock.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+
+"""
+Rotate to find a fiducial, and then navigate to a series of points
+
+It provides a service /rotate, which can be called from the command line:
+
+$ rosservice call /dock
```
Comment (rohbotics) [inline docking/nodes/dock.py:33]: Give the full command

**Review by rohbotics — APPROVED** [review]
Looks pretty good, I didn't spot anything that could cause the bug form yesterday, that was probably a fusion error.

File: docking/README.md (hunk 2 above)
Comment (jim-v) [reply docking/README.md]: Thanks for catching that.  The waypoints were parameters at one point, then became part of the arguments to the service call

File: docking/package.xml (hunk 1 above)
Comment (jim-v) [reply docking/package.xml]: Sorry!

**Comment by rohbotics** [comment]
Thanks for the changes.

When you merge please squash and merge.

**Pull request merged**